### PR TITLE
[#30] Set up test / code linting tools + add Travis continuous integration

### PR DESCRIPTION
**Pull request by krissy**
**Addresses issue #30 + some extra bits**
# What this does

Some general cleanup and setup for testing, code linting and continuous integration going forward. 
## Tasks
- [x] Clean out unused files
- [x] Add linting tool (rubocop) and clean-up code based on initial run
- [x] Add testing tools/configuration
- [x] Add factories for Drawing and User models
- [x] Add some initial tests for Drawing models/controllers to kickstart coverage (others can use this as a basis to add more tests)
- [x] Add some validations to the Drawings model
- [x] Add CI via travis (builds here: https://travis-ci.org/empowerhack/DrawMyLife-Service)
- [x] Add sloth

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -1,2 +1,3 @@
 --color
 --require spec_helper
+--format documentation
```
Comment (steverabino) [inline .rspec]: New line

File: Gemfile [hunk 2]
```diff
@@ -50,6 +49,8 @@ gem 'country_select'
 
 gem 'rack-cors', '~> 0.3.1'
 
+gem 'rubocop', '~> 0.42.0', require: false
```
Comment (steverabino) [inline Gemfile:52]: Comment for what it does

File: spec/controllers/drawings_controller_spec.rb [hunk 3]
```diff
@@ -1,5 +1,66 @@
...
+        expect{ perform }.to change{ Drawing.count }.by(1)
+      end
+
+      it "redirects to index" do
+        expect(perform).to redirect_to drawings_path
+      end
+    end
+
+    context "with invalid params" do
+      let(:params) { { description: "This is not enough" } }
```
Comment (steverabino) [inline spec/controllers/drawings_controller_spec.rb:21]: Why the double `{ {` ?

**Comment by steverabino** [comment]
Not much else I can add! If it passes, I'm happy! 🎉

File: spec/controllers/drawings_controller_spec.rb (hunk 3 above)
Comment (krissy) [inline spec/controllers/drawings_controller_spec.rb:21]: The outer pair of braces encloses a single-line block for the `let` and the inner pair is the param hash. Does look confusing sometimes!

**Comment by steverabino** [comment]
:+1:

**Pull request merged**